espressostreamer/espresso_streamer.go

```diff
@@ -326,29 +337,43 @@ func fetchNextHotshotBlock(
 	nextHotshotBlockNum uint64,
 	parseHotShotPayloadFn func(tx espressoTypes.Bytes) ([]*MessageWithMetadataAndPos, error),
 	namespace uint64,
+	parseTxPayloadRetryBackoff time.Duration,
 ) ([]*MessageWithMetadataAndPos, error) {
 	arbTxns, err := espressoClient.FetchTransactionsInBlock(ctx, nextHotshotBlockNum, namespace)
 	if err != nil {
-		return []*MessageWithMetadataAndPos{}, fmt.Errorf("%w: %w", FailedToFetchTransactionsErr, err)
+		return []*MessageWithMetadataAndPos{}, fmt.Errorf("%w: %w", ErrFailedToFetchTransactions, err)
 	}
 
 	result := []*MessageWithMetadataAndPos{}
 
 	for _, tx := range arbTxns.Transactions {
-		messages, err := parseHotShotPayloadFn(tx)
-		if err != nil {
-			log.Warn("failed to verify espresso transaction", "err", err)
-			continue
+		var retryCount uint64
+		for {
+			messages, err := parseHotShotPayloadFn(tx)
+			if err == nil {
+				result = append(result, messages...)
+				break
+			}
+
+			// If the error is Fatal (e.g., batch poster signature verification failed), we skip to the next transaction.
+			if ErrorIsFatal(err) {
+				log.Warn("Fatal error while parsing payload, skipping transaction", "err", err)
+				break
+			}
+
+			retryCount++
+			log.Warn("Retriable error while parsing payload, will retry", "err", err, "retryCount", retryCount)
+			// In the case of an rpc error, we retry after a delay.
+			time.Sleep(parseTxPayloadRetryBackoff)
```

Comment:
I don't think we need a for-loop here. I think we can just return error if the error is not `fatal`?